cmd/antrea-agent/options.go

```diff
@@ -514,87 +412,44 @@ func (o *Options) validateK8sNodeOptions() error {
 		}
 		o.dnsServerOverride = hostPort
 	}
-
-	if features.DefaultFeatureGate.Enabled(features.Egress) {
-		if o.config.Egress.MaxEgressIPsPerNode > defaultMaxEgressIPsPerNode {
-			return fmt.Errorf("maxEgressIPsPerNode cannot be greater than %d", defaultMaxEgressIPsPerNode)
-		}
-	}
+	o.enableAntreaPolicy = true
 	return nil
 }
 
-// resetVMDefaultFeatures sets the feature's default enablement status as false if it is not supported on a VM or a BM.
-func (o *Options) resetVMDefaultFeatures() error {
-	disabledFeatureMap := make(map[string]bool)
-	for f, s := range features.DefaultAntreaFeatureGates {
-		if s.Default && !features.SupportedOnExternalNode(f) {
-			disabledFeatureMap[string(f)] = false
-		}
+func (o *Options) validateCNIConfig() error {
+	cidrs, err := utilnet.ParseCIDRs(o.config.TransportInterfaceCIDRs)
+	if err != nil {
+		return fmt.Errorf("transportInterfaceCIDRs %v is invalid: %v", o.config.TransportInterfaceCIDRs, err)
```

Comment:
maybe we should use the `%w` formatter consistently for new "errors" (and edit the coding guide)?